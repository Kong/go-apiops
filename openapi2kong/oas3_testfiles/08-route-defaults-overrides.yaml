# x-kong-route-defaults can be specified and will be honored on each level

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/
x-kong-route-defaults:
  regex_priority: 100
  # routes cannot have a foreign key to a service (is set by the conversion)
  service: delete-me
paths:
  /path1:
    get:
      # should get the document level defaults
      operationId: uses-doc-defaults
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
  /path2:
    # specify new defaults to override document level
    x-kong-route-defaults:
      regex_priority: 200
      # routes cannot have a foreign key to a service (is set by the conversion)
      service: delete-me
    get:
      operationId: uses-path-defaults
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
    post:
      # specify new defaults to override path level
      x-kong-route-defaults:
        regex_priority: 300
        # routes cannot have a foreign key to a service (is set by the conversion)
        service: delete-me
      operationId: uses-ops-defaults
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
