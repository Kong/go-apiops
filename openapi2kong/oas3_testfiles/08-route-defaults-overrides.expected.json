{
  "_format_version": "3.0",
  "services": [
    {
      "host": "server1.com",
      "id": "0907c4ab-d9e4-5d21-813b-c57a97eeaad9",
      "name": "simple-api-overview",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "44777959-fdbe-5873-a5c1-beabbc822656",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-doc-defaults",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_08-route-defaults-overrides.yaml"
          ]
        },
        {
          "id": "a8cf87ef-dae0-5948-93e4-48f579fe12a0",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-path-defaults",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_08-route-defaults-overrides.yaml"
          ]
        },
        {
          "id": "5a98eef7-b0d2-572e-8656-4654a89c4179",
          "methods": [
            "POST"
          ],
          "name": "simple-api-overview_uses-ops-defaults",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_08-route-defaults-overrides.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_08-route-defaults-overrides.yaml"
      ]
    }
  ],
  "upstreams": []
}